### DISCUSSION TITLE
Why is Starlette dependency pinned to the minor version?

### DISCUSSION THREAD

**Discussion by kissgyorgy** [fastapi/fastapi#2532] [Questions] [answered]
Hello!

Is there a specific reason why Starlette is pinned to minor version? We would like to use multiple frameworks in the same process (e.g. ariadne for GraphQL API on a different endpoint), but we can't use the latest and greatest Starlette version because of FastAPI.

**Comment by Mause** [comment, marked as answer]
FastAPI is integrated quite closely with starlette, and unfortunately starlette doesn't use semantic versioning, so it's easy for a breaking starlette change to break FastAPI. The latest version of starlette does this, for example
